### PR TITLE
fix(Modal): modal confirm button should not be clicked when onOk method return promise

### LINKED ISSUE #37329: Modal.confirm这个方法用Promise控制延迟关闭时，快速点击是，当resolve的瞬间按钮loading消失的时机比弹窗销毁的时机快，导致按钮能再次点击到
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/que-ren-dui-hua-kuang-antd-4-22-8-forked-06k1t3?file=/demo.js)

### Steps to reproduce

1：点击按钮
2：点击确定
3：在loading状态的按钮上快速点击

### What is expected?

先销毁弹框再取消loading

### What is actually happening?

先取消loading了导致未销毁之前还能触发到按钮的点击事件

| Environment | Info |
| --- | --- |
| antd | 4.22.8 |
| React | 17.0.2 |
| System | win10 |
| Browser | 谷歌 |

### PR DESCRIPTION
**Pull request by KangXinzhi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #37329

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | modal confirm button should not be clicked when onOk method return promise |
| 🇨🇳 Chinese | 当确认方法返回promise后，modal confirm的确认按钮不应该可被再次点击 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/_util/ActionButton.tsx [hunk 1]
```diff
@@ -49,17 +49,25 @@ const ActionButton: React.FC<ActionButtonProps> = props => {
     setLoading(true);
     returnValueOfOnOk!.then(
       (...args: any[]) => {
-        setLoading(false, true);
         onInternalClose(...args);
         clickedRef.current = false;
+        // close方法是宏任务，为了防止关闭弹窗时再次点击，100ms后，再重置loading状态
+        // https://github.com/ant-design/ant-design/issues/37329
+        setTimeout(() => {
```
Comment (zombieJ) [inline components/_util/ActionButton.tsx:56]: timeout 太松散了，如果 motion 从 0.3s 调成 1s 就又会看到关闭过渡过程中变回来。这里需要处理的时候关闭时不再更新状态。

Comment (KangXinzhi) [reply components/_util/ActionButton.tsx]: > timeout 太松散了，如果 motion 从 0.3s 调成 1s 就又会看到关闭过渡过程中变回来。这里需要处理的时候关闭时不再更新状态。

ok，关闭时不在更新loading的状态了

File: components/_util/ActionButton.tsx [hunk 2]
```diff
@@ -49,16 +49,13 @@ const ActionButton: React.FC<ActionButtonProps> = props => {
...
-        setLoading(false, true);
         onInternalClose(...args);
         clickedRef.current = false;
       },
       (e: Error) => {
         // Emit error when catch promise reject
         // eslint-disable-next-line no-console
         console.error(e);
-        // See: https://github.com/ant-design/ant-design/issues/6183
-        setLoading(false, true);
```
Comment (afc163) [inline components/_util/ActionButton.tsx:61]: 这样会有问题，看看 https://github.com/ant-design/ant-design/issues/6183

**Comment by afc163** [comment]
换了种方式：https://github.com/ant-design/ant-design/pull/38400

**Pull request closed without merging**